src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_TABLE_INCREASING = 2;
+    private int threshold;
+    private int capacity;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        capacity = DEFAULT_INITIAL_CAPACITY;
+        threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        int index = calculateIndex(key);
+        Node<K,V> possibleNode = table[index];
+        if (possibleNode == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        while (possibleNode.next != null || Objects.equals(possibleNode.key, key)) {
+            if (Objects.equals(possibleNode.key, key)) {
+                possibleNode.value = value;
+                return;
+            }
+            possibleNode = possibleNode.next;
+        }
+        possibleNode.next = new Node<>(key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> nodeToSearch = table[calculateIndex(key)];
+        if (nodeToSearch == null) {
+            return null;
+        }
+        while (!Objects.equals(nodeToSearch.key, key)) {
```

Comment:
```suggestion
        while (nodeToSearch != null) {
            ...
```
let's try to simplify this method as well so that we won't need the upper check?

Reply:
Changed